mpire/pool.py

```diff
@@ -330,14 +381,28 @@ def stop_and_join(self) -> None:
                     break
                 self._restart_workers()
 
-            # Join workers
-            self.join()
+            # Obtain results from the exit results queues (should be done before joining the workers). When an error
+            # occurred inside the exit function we need to catch it here.
+            for wid, q in enumerate(self._exit_results_queues or []):
+                while not self._exception_thrown.is_set():
+                    try:
+                        self._exit_results.append(q.get(block=True, timeout=0.1))
+                        q.task_done()
+                        break
+                    except queue.Empty:
+                        pass
+                if self._exception_thrown.is_set():
+                    return
 
             # Join queues
             self._tasks_queue.join()
             self._results_queue.join()
+            [q.join() for q in self._exit_results_queues or []]
```

Comment:
If `_exit_results_queues` is initialized as an empty list you don't need to do the `or []` here. But this is dependent on another comment

Reply:
ok, ok :p I'll use the empty list then